Sources/IBLinterKit/Rules/ReuseIdentifier.swift

```diff
@@ -0,0 +1,58 @@
+import IBDecodable
+
+extension Rules {
+    
+    struct ReuseIdentifier: Rule {
+        
+        static let identifier = "reuse_identifier"
+        static let description = "Check that ReuseIdentifier same as class name."
+        
+        init(context: Context) {}
+        
+        func validate(storyboard: StoryboardFile) -> [Violation] {
+            guard let scenes = storyboard.document.scenes else { return [] }
+            let views = scenes.compactMap { $0.viewController?.viewController.rootView }
+            return views.flatMap { validate(for: $0, file: storyboard) }
+        }
+        
+        func validate(xib: XibFile) -> [Violation] {
+            // TODO: support Xib
+            return []
+        }
+        
+        private func validate<T: InterfaceBuilderFile>(for view: ViewProtocol, file: T) -> [Violation] {
+            let violation: [Violation] = {
+                func makeViolation(from view: ViewProtocol) -> Violation {
+                    let message = "\(viewName(of: view)) should have the same reuseidentifier"
+                    return Violation(pathString: file.pathString, message: message, level: .warning)
+                }
+                
+                // Currently only supported TableViewCell and CollectionViewCell (contains UICollectionReusableView)
+                if let tableView = view as? TableView {
+                    return tableView.prototypeCells?.compactMap {
+                        guard let customClass = $0.customClass, let reuseIdentifier = $0.reuseIdentifier else {
```

Comment:
there is a `IBReusable` protocol implemented by the two type of cell
maybe it could help to factorise some code

So you can extract a function or append list of cells into one before using `compactMap`
Or we could also add a protocol on `TableView` and `CollectionView` to provide the `reusableCells` of type `IBReusable`, so only one `if let parentResuable = view as? IBReusableParentView {` is necessary

Reply:
@phimage Thanks for your feedback. I'll try that pattern to aggregate logic :)

Reply:
by the way, that view types are defined in IBDecodable. Should I modify that project first or add extension it from IBLinter project?

Reply:
ok, firstly I try to append list of cell pattern without modify IBDecodable project.